AsyncDisplayKit/Layout/ASLayoutElement.h

```diff
@@ -245,6 +250,20 @@ extern NSString * const ASLayoutElementStyleLayoutPositionProperty;
  */
 @property (nonatomic, assign, readwrite) ASDimension maxWidth;
 
+#if YOGA
+@property (nonatomic, assign, readwrite) ASStackLayoutDirection direction;
```

Comment:
@maicki I decided to use compile-gated extensions in the base style object for a few reasons:

1) They are compiled out by default
2) Apps that use them will find them to be very performance-sensitive due to very widespread use
3) They aren't particularly feature-specific, but rather standard expressions of CSS Flexbox
4) The Extensibility PR had not landed by the time I started this, and isn't merging (also see my comment about ASEnvironment).

All that said, I would like to find a path to landing this diff soon that is "acceptable" and then I can actively improve it from there, so that the time-consuming merge conflicts we've all been dealing with are reduced.  For example, we could create a new style object like .yogaStyle, or a category .h / .mm for these new properties.